velox/connectors/hive/PartitionIdGenerator.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/PartitionIdGenerator.h"
+
+#include "velox/common/base/Portability.h"
+
+namespace facebook::velox::connector::hive {
+
+PartitionIdGenerator::PartitionIdGenerator(
+    const RowTypePtr& inputType,
+    std::vector<column_index_t> partitionChannels,
+    memory::MemoryPool* FOLLY_NONNULL pool)
+    : partitionChannels_(std::move(partitionChannels)),
+      allPartitionsVector_(std::static_pointer_cast<RowVector>(
+          BaseVector::create(inputType, 0, pool))),
+      activeIds_(SelectivityVector::empty(0)),
+      maxId_(0) {
+  hashers_.reserve(partitionChannels_.size());
+  for (column_index_t i = 0; i < partitionChannels_.size(); i++) {
+    hashers_.push_back(exec::VectorHasher::create(
+        inputType->childAt(partitionChannels_[i]), partitionChannels_[i]));
+  }
+}
+
+std::map<uint64_t, uint64_t> PartitionIdGenerator::run(
+    const RowVectorPtr& input,
+    raw_vector<uint64_t>& result) {
+  result.resize(input->size());
+  bool rehashed =
+      computeIds(input, SelectivityVector{input->size(), true}, result);
```

Comment:
nit: true is the default value, hence, it is sufficient to use `SelectivityVector(input->size())`.

nit: It would be nice to re-use SelectivityVector across batches by storing it in a member variable.

Reply:
Added a member variable allRows_, for reuse by input.

Here I call `allRows_.resize(input->size());` and `allRows_.setAll();` with `allRows_.setAll();` set the `allSelected_` field to true for fast access.